compile.sh

```diff
@@ -26,4 +34,8 @@ swiftc -O -parse-as-library -Xcc -funroll-loops -Xcc -march=native -Xcc -ftree-v
 # haxe --class-path haxe -main Code --jvm haxe/code.jar # was getting errors running `haxelib install hxjava`
 #dotnet publish csharp/csharp.csproj -o csharp/code-aot /p:PublishAot=true
 dotnet publish csharp/csharp.csproj -o csharp/code
-ghc -O2 -fllvm haskell/code.hs -o haskell/code || { echo "ghc: cannot compile with llvm backend; fallback to use default backend"; ghc -O2 haskell/code.hs -o haskell/code; }
+cabal update
+cabal install --lib random --force
+ghc -O2 -fllvm -Wno-partial-fields haskell/code.hs -o haskell/code || { echo "ghc: cannot compile with llvm backend; fallback to use default backend"; ghc -O2 -Wno-partial-fields haskell/code.hs -o haskell/code; }
```

Comment:
If we're using llvm, we should also tell it to optimise its results.

```suggestion
ghc -O2 -fllvm -optlo -O3 -Wno-partial-fields haskell/code.hs -o haskell/code || { echo "ghc: cannot compile with llvm backend; fallback to use default backend"; ghc -O2 -Wno-partial-fields haskell/code.hs -o haskell/code; }
```